townhall/myapi/serializers.py

```diff
@@ -12,6 +12,16 @@ class Meta:
         fields = "__all__"
 
 
+class FilteredOpportunitySerializer(serializers.Serializer):
+    title = serializers.CharField(required=False)
```

Comment:
I believe instead of using it in serializer, filter can be abstracted to its own layer (read up here https://django-filter.readthedocs.io/en/stable/ref/filterset.html) 

I think its worth a refactor to use FilterSet or else serializer will get messy with filtering mechanisms.